### PR TITLE
Log obfuscated json envelope on json validation failure

### PR DESCRIPTION
**Pull request by jaceko**

### CODE REVIEW

File: messaging/messaging-core/src/main/java/uk/gov/justice/services/messaging/JSONObjectValueObfuscator.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+     * @param json
+     * @return new json with obfuscated values
+     */
+    public static JSONObject obfuscated(final JSONObject json) {
+        final JSONObject result = new JSONObject();
+        obfuscate(json, result);
+        return result;
+    }
+
+    private static void obfuscate(final JSONObject in, final JSONObject out) {
```
Comment (mapingo) [inline messaging/messaging-core/src/main/java/uk/gov/justice/services/messaging/JSONObjectValueObfuscator.java]: in and out arguments could be more descriptive

File: messaging/messaging-core/src/main/java/uk/gov/justice/services/messaging/JSONObjectValueObfuscator.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        in.keys().forEachRemaining(key -> {
+            JSONObject nestedJson = in.optJSONObject(key);
```
Comment (mapingo) [inline messaging/messaging-core/src/main/java/uk/gov/justice/services/messaging/JSONObjectValueObfuscator.java]: nestedJson can be final

File: messaging/messaging-core/src/main/java/uk/gov/justice/services/messaging/JSONObjectValueObfuscator.java [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+                out.put(key, newJsonArray);
+
+            } else {
+                out.put(key, obfuscatedValueOf(in.get(key)));
+            }
+
+        });
+    }
+
+    private static boolean isArray(final String key, final JSONObject in) {
```
Comment (mapingo) [inline messaging/messaging-core/src/main/java/uk/gov/justice/services/messaging/JSONObjectValueObfuscator.java]: in argument could be more descriptive

File: messaging/messaging-core/src/main/java/uk/gov/justice/services/messaging/JSONObjectValueObfuscator.java [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+        } catch (Exception e) {
+            return false;
+        }
+    }
+
+    private static boolean isUuid(final Object object) {
+        return object instanceof UUID;
+    }
+
+    private static Object obfuscatedValueOf(Object object) {
```
Comment (mapingo) [inline messaging/messaging-core/src/main/java/uk/gov/justice/services/messaging/JSONObjectValueObfuscator.java]: object arg can be final

File: messaging/messaging-core/src/test/java/uk/gov/justice/services/messaging/JSONObjectValueObfuscatorTest.java [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+        with(obfuscated(new JSONObject()
+                .put("property1", randomUUID())
+                .put("nested", new JSONObject().put("property2", randomUUID())))
+                .toString())
+                .assertThat("property1", is("xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"))
+                .assertThat("nested.property2", is("xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"));
+    }
+
+    @Test
+    public void shouldReplaceValuesInArray() throws Exception {
```
Comment (mapingo) [inline messaging/messaging-core/src/test/java/uk/gov/justice/services/messaging/JSONObjectValueObfuscatorTest.java]: Could have test for a JSONArray containing a JSONObject and a JSONArray as well

File: messaging/messaging-core/src/main/java/uk/gov/justice/services/messaging/JSONObjectValueObfuscator.java (hunk 1 above)
Comment (jaceko) [inline messaging/messaging-core/src/main/java/uk/gov/justice/services/messaging/JSONObjectValueObfuscator.java]: done

File: messaging/messaging-core/src/main/java/uk/gov/justice/services/messaging/JSONObjectValueObfuscator.java (hunk 2 above)
Comment (jaceko) [inline messaging/messaging-core/src/main/java/uk/gov/justice/services/messaging/JSONObjectValueObfuscator.java]: done

File: messaging/messaging-core/src/main/java/uk/gov/justice/services/messaging/JSONObjectValueObfuscator.java (hunk 3 above)
Comment (jaceko) [inline messaging/messaging-core/src/main/java/uk/gov/justice/services/messaging/JSONObjectValueObfuscator.java]: done

File: messaging/messaging-core/src/main/java/uk/gov/justice/services/messaging/JSONObjectValueObfuscator.java (hunk 4 above)
Comment (jaceko) [inline messaging/messaging-core/src/main/java/uk/gov/justice/services/messaging/JSONObjectValueObfuscator.java]: done

File: messaging/messaging-core/src/test/java/uk/gov/justice/services/messaging/JSONObjectValueObfuscatorTest.java (hunk 5 above)
Comment (jaceko) [inline messaging/messaging-core/src/test/java/uk/gov/justice/services/messaging/JSONObjectValueObfuscatorTest.java]: done

**Pull request merged**